File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        String filteredStr = s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
        
        if (filteredStr.isEmpty()){
            return true;
        }

        int len = filteredStr.length();
        
        int ptr1 = (len / 2) - 1;
        int ptr2 = len % 2 == 0 ? len / 2 : len / 2 + 1;
        
        while(ptr1 >= 0 && ptr2 < len){
            if(filteredStr.charAt(ptr1) != filteredStr.charAt(ptr2)){
                return false;
            }
            
            ptr1 -= 1;
            ptr2 += 1;
        }
        
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        filterStr = re.sub(r"[^A-Za-z0-9]", "", s)
        
        if not filterStr:
            return True
        
        length = len(filterStr)
        
        ptr1 = length // 2 - 1
        ptr2 = length // 2 if length % 2 == 0 else length // 2 + 1
        
        
        while(ptr1 >= 0 and ptr2 < length):
            
            if filterStr[ptr1].lower() != filterStr[ptr2].lower():
                return False
            
            ptr1 -= 1
            ptr2 += 1
            
            
        return True
            
        
        